src/ViewRenderer/ViewRenderer.php

```diff
@@ -108,37 +104,48 @@ public function withLayout(string $layout): self
         return $new;
     }
 
-    public function withCsrf(string $requestAttribute = Csrf::REQUEST_NAME): self
+    /**
+     * @param ContentParamsInjectionInterface[]|LayoutParamsInjectionInterface[]|LinkTagsInjectionInterface[]|MetaTagsInjectionInterface[] $injections
+     * @return self
+     */
+    public function addInjections(array $injections): self
     {
         $new = clone $this;
-        $new->csrfTokenRequestAttribute = $requestAttribute;
-        $new->csrfToken = $new->getCsrfToken();
+        $new->injections = array_merge($this->injections, $injections);
+        return $new;
+    }
 
+    /**
+     * @param ContentParamsInjectionInterface|LayoutParamsInjectionInterface|LinkTagsInjectionInterface|MetaTagsInjectionInterface $injection
+     * @return self
+     */
+    public function addInjection($injection): self
+    {
+        return $this->addInjections([$injection]);
+    }
+
+    /**
+     * @param ContentParamsInjectionInterface[]|LayoutParamsInjectionInterface[]|LinkTagsInjectionInterface[]|MetaTagsInjectionInterface[] $injections
+     * @return self
+     */
+    public function withInjections(array $injections): self
+    {
+        $new = clone $this;
+        $new->injections = $injections;
         return $new;
     }
 
     private function renderProxy(string $view, array $parameters = []): string
     {
-        if ($this->csrfToken !== null) {
-            $parameters['csrf'] = $this->csrfToken;
-            $this->view->registerMetaTag(
-                [
-                    'name' => 'csrf',
-                    'content' => $this->csrfToken,
-                ],
-                'csrf_meta_tags'
-            );
-        }
+        $parameters = $this->injectContent($parameters, $this->injections);
```

Comment:
Maybe you name this method `injectContentParameters`?

Reply:
In this method inject not only parameters, but also meta- and link-tags.

Reply:
Done.